modules/images/picture-element/load.php

```diff
@@ -0,0 +1,130 @@
+ *
+ * @return string The new image tag.
+ */
+function wrap_image_in_picture( $image, $attachment_id ) {
+	$image_meta              = wp_get_attachment_metadata( $attachment_id );
+	$original_file_mime_type = get_post_mime_type( $attachment_id );
+	if ( false === $original_file_mime_type ) {
+		return false;
+	}
+
+	// Collect all of the sub size image mime types.
+	$sub_size_mime_types_heap = array();
+	foreach ( $image_meta['sizes'] as $size ) {
+		$sub_size_mime_types_heap[ $size['mime-type'] ] = true;
+	}
+	$sub_size_mime_types = array_keys( $sub_size_mime_types_heap );
+
+	/**
+	 * Filter the image mime types that can be used for the <picture> element.
+	 *
+	 * Default is: ['image/webp'] for testing. Returning an empty array will skip using the picture element.
+	 *
+	 * The mime types will output in the picture element in the order they are provided.
+	 * The original image will be used as the fallback.
+	 *
+	 * @since n.e.x.t
+	 * @param string[] mime types than can be used.
+	 * @param int $attachment_id The id of the image being evaluated.
+	 */
+	$enabled_mime_types = apply_filters(
+		'wp_picture_element_mime_types',
+		array(
+			'image/webp',
+		),
+		$attachment_id
+	);
+
+	$mime_types = array_filter(
+		$sub_size_mime_types,
+		function( $mime_type ) use ( $enabled_mime_types ) {
+			return in_array( $mime_type, $enabled_mime_types, true );
+		}
+	);
+
+	// No eligible mime types.
+	if ( ! $mime_types ) {
+		return false;
+	}
+
+	// If the original mime types is the only one available, no picture element is needed.
+	if ( 1 === count( $mime_types ) && $mime_types[0] === $original_file_mime_type ) {
+		return false;
+	}
+
+	// Add each mime type to the picture's sources.
+	$picture_sources              = '';
+	$image                        = wp_get_attachment_image_src( $attachment_id, 'full', false );
+	list( $src, $width, $height ) = $image;
+	$size_array                   = array( absint( $width ), absint( $height ) );
+
+	foreach ( $mime_types as $image_mime_type ) {
+		// @TODO limit by mime type when multiple mime types are supported.
+		// @TODO use sizes array, not medium.
+		$sizes            = wp_calculate_image_sizes( $size_array, $src, $image_meta, $attachment_id );
+		$image_srcset     = wp_get_attachment_image_srcset( $attachment_id, $size_array, $image_meta );
+		$picture_sources .= sprintf(
+			'<source type="%s" srcset="%s" sizes="%s">',
+			$image_mime_type,
+			$image_srcset,
+			$sizes
+		);
+	}
+
+	// Fall back to the original image without a srcset.
+	$original_sizes = array( $image[1], $image[2] );
+	$original_image = wp_get_original_image_url( $attachment_id );
+	add_filter( 'wp_calculate_image_srcset_meta', '__return_false' );
+	$original_image_without_srcset = wp_get_attachment_image( $attachment_id, $original_sizes, false, array( 'src' => $original_image ) );
+	remove_filter( 'wp_calculate_image_srcset_meta', '__return_false' );
+
+	return sprintf(
+		'<picture>%s %s</picture>',
```

Comment:
Might be useful to add a class here like the image tag. 

So 
```
<img class"wp-image-123" 
```

```
<picture class"wp-picture-123" 
```

This class that contains an id is extremely useful.

Reply:
added